tests/inputFields.spec.ts

```diff
@@ -1,94 +1,96 @@
+test.beforeEach(async ({ page }) => {
+    await page.goto('/')
+    await page.getByText('Pet Types').click()
 
- await expect(page.locator('h2')).toHaveText("Pet Types")
+    await expect(page.getByRole("heading")).toHaveText("Pet Types")
 })
 
 //3. Click on "Edit" button for the "cat" pet type
-test('change the name of the cat', async({page}) => {
-    const petTypesCat= page.locator('button').filter({hasText:"Edit"}).first()
-    await petTypesCat.click()
+test('change the name of the cat', async ({ page }) => {
+    const petTypesCatEditButton = page.getByRole('button', { name: "Edit" }).first()
+    await petTypesCatEditButton.click()
 
-    /*
-    Add assertion of the "Edit Pet Type" text displayed*/
 
-    await expect (page.locator('h2')).toHaveText('Edit Pet Type')
+    //4. Add assertion of the "Edit Pet Type" text displayed
+    await expect(page.getByRole("heading")).toHaveText('Edit Pet Type')
 
 
-//Change the pet type name from "cat" to "rabbit" and click "Update" button
-const correctPetType = page.locator('#name')
-await correctPetType.click()
-await correctPetType.clear()
-await correctPetType.fill('rabbit')
-await page.locator('button',{hasText:'Update'}).click()
+    //5. Change the pet type name from "cat" to "rabbit" and click "Update" button
+    const editPetTypeInputField = page.locator('#name')
+    await editPetTypeInputField.click()
+    await editPetTypeInputField.clear()
+    await editPetTypeInputField.fill('rabbit')
+    await page.getByRole('button', { name: 'Update' }).click()
 
-//6. Add the assertion that the first pet type in the list of types has a value "rabbit" 
-await expect(page.locator('input[name=pettype_name]').first()).toHaveValue('rabbit')
+    //6. Add the assertion that the first pet type in the list of types has a value "rabbit" 
+    await expect(page.locator('input[name=pettype_name]').first()).toHaveValue('rabbit')
 
 
-//7. Click on "Edit" button for the same "rabbit" pet type
+    //7. Click on "Edit" button for the same "rabbit" pet type
 
-await petTypesCat.click()
+    await petTypesCatEditButton.click()
 
-//8. Change the pet type name back from "rabbit" to "cat" and click "Update" button
+    //8. Change the pet type name back from "rabbit" to "cat" and click "Update" button
 
-await correctPetType.click()
-await correctPetType.clear()
-await correctPetType.fill('cat')
-await page.locator('button', {hasText: 'Update'}).click()
+    await editPetTypeInputField.click()
+    await editPetTypeInputField.clear()
+    await editPetTypeInputField.fill('cat')
+    await page.getByRole('button', { name: 'Update' }).click()
 
-//9. Add the assertion that the first pet type in the list of names has a value "cat" 
-await expect(correctPetType).toHaveValue('cat')
+    //9. Add the assertion that the first pet type in the list of names has a value "cat" 
+    const firstPetType = page.locator('input[name=pettype_name]').first()
+    await expect(firstPetType).toHaveValue('cat')
```

Comment:
If you decided to create a firstPetType constant for this locator, then need to declare it earlier, since the same locator is used in line 28. Then in both assertions you will use a reference to the locator